contributors/design-proposals/service-injection-policy.md

```diff
@@ -0,0 +1,279 @@
+# Service Injection Policy
+
+## Abstract
+
+Describes a policy resource that allows loose coupling from services to the pods
+that consume them.
+
+## Motivation
+
+Consuming a service involves more than just connectivity.  In addition to
+coordinates to reach the service, credentials and non-secret configuration
+parameters are typically needed to use the service.  The primitives for this
+already exist, but a gap exists where loose coupling is desired: it should be
+possible to inject pods with the information they need to use a service on a
+service-by-service basis, without the pod authors having to incorporate the
+information into every pod spec where it is needed.
+
+## Constraints and Assumptions
+
+1.  New mechanisms must be made to work with controllers such as deployments and
+    replicasets that create pods. Existing controllers that create pods should
+    recreate their pods when a new Service Injection Policy is added that would
+    effect them.
+
+## Use Cases
+
+1.  As a user, I want to be able to describe a way that pods should be injected
+    with the information to consume a particular service in a loosely-coupled
+    way, so that I can concisely model the information about how the service
+    should be consumed without altering every consuming pod spec.
+
+### Loose coupling between services and their consumers
+
+Using a Service Injection Policy allows pod authors to not have to explicitly
+set information for every pod. The information will be injected based off the
+policy's `Selector`.
+
+## Proposed Changes
+
+### ServiceInjectionPolicy API object
+
+```go
+type ServiceInjectionPolicy struct {
+    unversioned.TypeMeta
+    ObjectMeta
+
+    Spec ServiceInjectionPolicySpec
+}
+
+type ServiceInjectionPolicySpec struct {
+    // Service is a reference to the service being injected (optional, immutable).
+    Service       string
```

Comment:
@derekwaynecarr we chatted about some use-cases you had related to resource consumption -- are those solidified at all?  We are trying to determine whether this is really a generic injection policy and not intrinsically related to services.